operatorcourier/errors.py

```diff
@@ -0,0 +1,46 @@
+"""
+operatorcourier.errors
+
+Defines custom operator-courier exceptions.
+"""
+
+
+class OpCourierError(ValueError):
+    """Base class for operator-courier exceptions,
+    subclassed from ValueError for backward compatibility
+    """
+    pass
+
+
+class OpCourierBadYaml(OpCourierError):
+    """Invalid yaml file"""
+    pass
+
+
+class OpCourierBadArtifact(OpCourierError):
+    """File is a valid yaml, but not a valid CSV, CRD or Package"""
+    pass
+
+
+class OpCourierBadBundle(OpCourierError):
+    """Invalid bundle (e.g. missing CSV/CRD/Package).
+    Contains the info collected during validation of the bundle.
+    """
+    def __init__(self, msg, validation_dict):
+        """
+        :param msg: the message of this exception
+        :param validation_dict: info collected during validation of bundle
+        """
+        super().__init__(msg)
+        self.validation_dict = validation_dict
+
+
+class OpCourierQuayError(OpCourierError):
+    """An error occurred while pushing bundle to Quay.io"""
+    def __init__(self, msg, code=500):
```

Comment:
Why do we have a default here? If there is an error returned from the server, we should always have an error code (even if the error is that the server is not available -- which is not 500).

Reply:
That's fair. Will remove default value.

Reply:
I was thinking about this, I'd really like to get content from quay what is wrong with error code and quay error JSON if possible.

We may actually split this into more exceptions:
* OpCourierQuayError -- generic quay error
* OpCourierQuayCommunicationError(OpCourierQuayError) - random error that happened during communication
*OpCourierQuayErrorResponse(OpCourierQuayError) -- error returned from quay with quay details, basically catching HTTPException from requests.

Reply:
@chmeliik This got +1 from Kevin

Reply:
Understood, will get on it tomorrow